### PR TITLE
Allow custom index name in the REST API

### PR DESCRIPTION
**Pull request by kikkauz**
As described in the ticket I simply added an input param to the rest endpoint to override the setting value, is the index param is not null.

### CODE REVIEW

**Comment by dadoonet** [comment]
I think it's a good thing to merge. I don't see a lot of complication with this as this is purely optional.
May I ask you to add some information about this in the documentation? Here: https://github.com/dadoonet/fscrawler/blob/master/docs/source/admin/fs/rest.rst

And may be a new integration test in https://github.com/dadoonet/fscrawler/blob/master/integration-tests/it-common/src/main/java/fr/pilato/elasticsearch/crawler/fs/test/integration/FsCrawlerRestIT.java ?

**Comment by kikkauz** [comment]
Great thanks! I'm quite busy today and tomorrow but I'll try to find some time for that.

**Comment by kikkauz** [comment]
Hello
I tried to implement a test basically copying and changing testAllDocumentsWithRest and some other stuff but I do not manage to make it work :-(
I'm not sure I'm doing the right thing

**Comment by dadoonet** [comment]
> I tried to implement a test basically copying and changing testAllDocumentsWithRest and some other stuff but I do not manage to make it work :-(

Could you push your changes in your branch so I can look?

**Comment by kikkauz** [comment]
I pushed the code. I'm sure it's a stupid error :-)

**Comment by dadoonet** [comment]
I don't understand why you are saying it does not work.
I mean that Travis seems to be happy with the changes and this new test.

**Comment by dadoonet** [comment]
Could you also apply the changes I proposed in #716?

**Comment by kikkauz** [comment]
interesting...then I guess it's my environment having problems....test was failing with message
1> 15:39:22,861 INFO  [f.p.e.c.f.t.f.AbstractFSCrawlerTestCase]   ---> Waiting up to 2m for 24 documents in fscrawler_custom
  1> 15:41:23,403 WARN  [f.p.e.c.f.t.f.AbstractFSCrawlerTestCase]      ---> expecting [24] but got [0] documents in fscrawler_custom

**Comment by dadoonet** [comment]
Are you running the tests from the IDE or from the command line?

File: docs/source/admin/fs/rest.rst [hunk 1]
```diff
@@ -251,6 +251,18 @@ The field ``external`` doesn't necessarily be a flat structure. This is a more a
 
 .. attention:: Only standard :ref:`FSCrawler fields <generated_fields>` can be set outside ``external`` field name.
 
+
+Specifying an elasticsearch index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+By default, fscrawler creates document in the index defined in the _settings.yaml (or json) file.
```
Comment (dadoonet) [inline docs/source/admin/fs/rest.rst]: Could you add a blank line just before this paragraph?

Comment (dadoonet) [reply docs/source/admin/fs/rest.rst]: Could you change the sentence from `defined in the _settings.yaml (or json) file.` to `defined in the ``_settings.yaml`` file.`.

**Comment by dadoonet** [comment]
Awesome. Thank you for contributing your change!

**Pull request merged**